zanata-model/src/main/java/org/zanata/model/validator/UniqueValidator.java

```diff
@@ -55,6 +57,10 @@ public boolean isValid(Object value, ConstraintValidatorContext context) {
     }
 
     private int countRows(Object value) {
+        // we need to use entityManager.unwrap because  injected session will
+        // be a weld proxy and criteria.getExecutableCriteria method will try
+        // to cast it to SessionImplementor (ClassCastException)
+        Session session = entityManager.unwrap(Session.class);
```

Comment:
@huangp Glad you got to the bottom of it. I think we should deprecate the session producer in EntityManagerProducer with a TODO comment warning about this problem, and maybe get rid of it in `master` in a separate PR.